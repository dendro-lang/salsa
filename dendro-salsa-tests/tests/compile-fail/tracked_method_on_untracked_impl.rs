#[salsa::jar(db = Db)]
struct Jar(MyInput, tracked_method_on_untracked_impl);

trait Db: salsa::DbWithJar<Jar> {}

#[salsa::input]
struct MyInput {
    field: u32,
}

impl MyInput {
    #[salsa::tracked]
    fn tracked_method_on_untracked_impl(self, db: &dyn Db) -> u32 {
        input.field(db)
    }
}

fn main() {}
