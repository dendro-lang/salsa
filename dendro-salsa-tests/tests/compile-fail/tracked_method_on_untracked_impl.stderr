error: #[salsa::tracked] must also be applied to the impl block for tracked methods
  --> tests/compile-fail/tracked_method_on_untracked_impl.rs:13:41
   |
13 |     fn tracked_method_on_untracked_impl(self, db: &dyn Db) -> u32 {
   |                                         ^^^^

error[E0412]: cannot find type `tracked_method_on_untracked_impl` in this scope
 --> tests/compile-fail/tracked_method_on_untracked_impl.rs:2:21
  |
2 | struct Jar(MyInput, tracked_method_on_untracked_impl);
  |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope
